### PR TITLE
[UI] Show cached steps

### PR DESCRIPTION
**Pull request by Bobgy**
Fixes https://github.com/kubeflow/pipelines/issues/2904
Demo: https://youtu.be/Zb4CA1G8dxE

### CODE REVIEW

**Comment by Bobgy** [comment]
@Ark-kun can you lgtm?
I've added required tests

File: frontend/src/lib/StatusUtils.test.tsx [hunk 1]
```diff
@@ -130,4 +135,59 @@ describe('StatusUtils', () => {
...
+          ...DEFAULT_NODE_STATUS,
+          phase: 'Succeeded',
+        }),
+      ).toEqual('Succeeded');
+    });
+
+    it('returns cached phase for a cached node', () => {
+      expect(
+        parseNodePhase({
+          ...DEFAULT_NODE_STATUS,
```
Comment (Ark-kun) [inline frontend/src/lib/StatusUtils.test.tsx:176]: Are you sure this status is correct?

File: frontend/src/lib/StatusUtils.test.tsx [hunk 2]
```diff
@@ -130,4 +135,59 @@ describe('StatusUtils', () => {
...
+          phase: 'Succeeded',
+        }),
+      ).toEqual('Succeeded');
+    });
+
+    it('returns cached phase for a cached node', () => {
+      expect(
```
Comment (Ark-kun) [inline frontend/src/lib/StatusUtils.test.tsx:174]: Is this test complete?

File: frontend/src/lib/StatusUtils.test.tsx (hunk 1 above)
Comment (Bobgy) [reply frontend/src/lib/StatusUtils.test.tsx:176]: Moved phase explicitly here (the ... syntax means expanding that object, and we override some attributes).

File: frontend/src/lib/StatusUtils.test.tsx (hunk 2 above)
Comment (Bobgy) [reply frontend/src/lib/StatusUtils.test.tsx:174]: Thanks! Completed it

**Pull request merged**